src/hotspot/share/opto/superword.cpp

```diff
@@ -2486,6 +2486,16 @@ VStatus VLoopBody::construct() {
   return VStatus::make_success();
 }
 
+static bool can_subword_truncate(Node* in) {
+  if (in->is_Load() || in->is_Store() || in->is_Convert()) {
+    return true;
+  }
+
+  int opc = in->Opcode();
+  return opc == Op_AddI || opc == Op_SubI || opc == Op_MulI || opc == Op_AndI || opc == Op_OrI || opc == Op_XorI
+    || opc == Op_ReverseBytesS || opc == Op_ReverseBytesUS;
```

Comment:
A switch might look nicer here, and be easier to extend later on ;)

Reply:
This list is a little scary... how do we know that we have all cases in it, and we are not getting regressions because we are missing some?

Reply:
A switch is a good idea, it'll definitely make the code easier to read. I've made the change in the recent commit. As for the cases, I ended up running the compiler unit tests and modifying the list until there were no test failures. Since we check for nodes that do not need truncation handling, any other nodes will automatically default to require truncation handling and fall back to not vectorizing.